Rubberduck.Parsing/VBA/BuiltInDeclarationLoader.cs

```diff
@@ -10,29 +10,19 @@ namespace Rubberduck.Parsing.VBA
     public class BuiltInDeclarationLoader : IBuiltInDeclarationLoader
     {
         private readonly IEnumerable<ICustomDeclarationLoader> _customDeclarationLoaders;
-        private RubberduckParserState _state;
+        private readonly RubberduckParserState _state;
 
         private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
 
         public BuiltInDeclarationLoader(RubberduckParserState state, IEnumerable<ICustomDeclarationLoader> customDeclarationLoaders)
         {
-            if (state == null)
-            {
-                throw new ArgumentNullException(nameof(state));
-            }
-            if (customDeclarationLoaders == null)
-            {
-                throw new ArgumentNullException(nameof(customDeclarationLoaders));
-            }
+            _state = state ?? throw new ArgumentNullException(nameof(state));
```

Comment:
IIRC it was exactly this change, that got me a deserved review-comment on my CodeMetrics PR. `ArgumentNullExceptions` are not generally collapsed in the codebase for preference reasons